File: app/src/main/java/com/studyasist/data/local/entity/ActivityType.kt
package com.studyasist.data.local.entity

import androidx.room.TypeConverter

enum class ActivityType {
    STUDY,
    BREAK,
    SCHOOL,
    TUITION,
    SLEEP
}

class ActivityTypeConverter {
    @TypeConverter
    fun from(value: ActivityType): String = value.name

    @TypeConverter
    fun to(value: String): ActivityType = ActivityType.valueOf(value)
}

File: StudyAsist-iOS/StudyAsist/Models/ActivityType.swift
//
//  ActivityType.swift
//  StudyAsist
//

import Foundation

enum ActivityType: String, Codable, CaseIterable {
    case study = "STUDY"
    case breakType = "BREAK"
    case school = "SCHOOL"
    case tuition = "TUITION"
    case sleep = "SLEEP"
}
